src/puppetlabs/trapperkeeper/services/status/status_core.clj

```diff
@@ -149,9 +161,17 @@
 
 (schema/defn ^:always-validate get-jvm-metrics :- JvmMetricsV1
   []
-  (let [runtime-bean (ManagementFactory/getRuntimeMXBean)]
+  (let [runtime-bean (ManagementFactory/getRuntimeMXBean)
+        gc-beans (jmx/mbean-names "java.lang:name=*,type=GarbageCollector")]
     {:heap-memory (jmx/read "java.lang:type=Memory" :HeapMemoryUsage)
      :non-heap-memory (jmx/read "java.lang:type=Memory" :NonHeapMemoryUsage)
+     :file-descriptors (clojure.set/rename-keys
+                        (jmx/read "java.lang:type=OperatingSystem" [:OpenFileDescriptorCount :MaxFileDescriptorCount])
+                        {:OpenFileDescriptorCount :used :MaxFileDescriptorCount :max})
+     :gc-stats (map #(conj {:name (.getCanonicalName %)}
```

Comment:
Looks like CollectionCount and CollectionTime are two attributes that _must_ be present if a MBean implements `java.lang:type=GarbageCollector`:

https://docs.oracle.com/javase/8/docs/api/java/lang/management/GarbageCollectorMXBean.html